src/server/components/lists/helpers.ts

```diff
@@ -11,12 +11,31 @@ import {
   fcdoLawyersPagesByCountry,
   listOfCountriesWithLegalAid,
 } from "server/services/metadata";
+import { URLSearchParams } from "url";
 
 export async function initLists(server: Express): Promise<void> {
   server.use(listsRouter);
 }
 
-export function queryStringFromParams(params: { [name: string]: any }): string {
+/**
+ * To support the select all option for checkbox fields, this function detects if the value "All" is provided and will
+ * remove all other options in the event "Select All" and any additional checkboxes were selected by the user.  Currently
+ * this is only required for lawyers.practiceArea (Areas of law field).
+ * @param params
+ */
+export function preProcessParams(params: { [name: string]: any}): { [name: string]: any} {
+  const paramsCopy = JSON.parse(JSON.stringify(params));
+
+  if (paramsCopy.practiceArea !== undefined) {
+    const areasOfLaw = paramsCopy.practiceArea;
+    if (Array.isArray(areasOfLaw) && areasOfLaw.some(item => item === "All")) {
+      paramsCopy.practiceArea = ["All"];
+    }
+  }
+  return paramsCopy;
+}
+
+export function queryStringFromParams(params: { [name: string]: any}, removeEmptyValues?: boolean): string {
```

Comment:
From what I can see it looks like there are two differences: 
  1. if the start value is a comma it takes a substring from index 1
  2. the content of arrays is flattened using .toString().  This differs to native node QueryString where an array would appear as myArray=1&myArray=2 rather than myArray=1,2 with queryStringFromParams

Could probably do with some refactoring to work out where these differences are in the code and change it to use native node QueryString at some point after the release.

Reply:
marking resolved per thumbs up :-)